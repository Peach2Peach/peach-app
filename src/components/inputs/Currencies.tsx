import React, { ReactElement } from 'react'
import { View } from 'react-native'
import { Checkboxes, Headline, Text } from '..'
import { CURRENCIES } from '../../constants'
import tw from '../../styles/tailwind'
import i18n from '../../utils/i18n'

type CurrenciesProps = {
  title: string,
  currencies: Currency[],
  setCurrencies: (currencies: Currency[]) => void
}

export default ({ title, currencies, setCurrencies }: CurrenciesProps): ReactElement => {
  const checkboxItems = CURRENCIES.map(value => ({
    value,
    display: <Text style={tw`-mt-0.5`}>
      {i18n(`currency.${value}`)} <Text style={tw`text-grey-1`}>({value})</Text>
    </Text>
  }))
  const onChange = (values: (string | number)[]) => values.length ? setCurrencies(values as Currency[]) : null

  return <View>
    <Headline style={tw`mt-16 text-grey-1`}>
      {title}
    </Headline>
    <Checkboxes
      style={tw`px-7 mt-2`}
      items={checkboxItems}
      selectedValues={currencies}
      onChange={onChange}/>
  </View>
}