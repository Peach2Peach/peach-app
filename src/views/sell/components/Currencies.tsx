import React, { ReactElement } from 'react'
import { View } from 'react-native'
import { Checkboxes, Headline, Text } from '../../../components'
import { CURRENCIES } from '../../../constants'
import tw from '../../../styles/tailwind'
import i18n from '../../../utils/i18n'

type CurrenciesProps = {
  currencies: Currency[],
  setCurrencies: (currencies: Currency[]) => void
}

export default ({ currencies, setCurrencies }: CurrenciesProps): ReactElement => <View>
  <Headline style={tw`mt-9`}>
    {i18n('sell.currencies')}
  </Headline>
  <Checkboxes
    style={tw`px-7 mt-2`}
    items={CURRENCIES.map(value => ({
      value,
      display: <Text>
        {i18n(`currency.${value}`)} <Text style={tw`text-grey-1`}>({value})</Text>
      </Text>
    }))}
    selectedValues={currencies}
    onChange={values => setCurrencies(values as Currency[])}/>
</View>