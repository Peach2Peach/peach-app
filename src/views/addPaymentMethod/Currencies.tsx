import { PeachScrollView, RadioButtons } from '../../components'
import { CURRENCIES } from '../../paymentMethods'
import { CurrencyType } from '../../store/offerPreferenes/types'
import tw from '../../styles/tailwind'
import i18n from '../../utils/i18n'
import { getCurrencyTypeFilter } from './utils'

const getDisplayName = (c: Currency) => {
  if (c === 'USDT') return i18n(`currency.${c}`)
  if (c === 'SAT') return i18n('paymentMethod.lnurl')
  return i18n(`currency.${c}`)
}

type Props = {
  currency: Currency
  setCurrency: (c: Currency) => void
  type: CurrencyType
}

export const Currencies = ({ currency, setCurrency, type }: Props) => {
  const currencies = CURRENCIES.filter(getCurrencyTypeFilter(type)).map((c) => ({
    value: c,
    display: getDisplayName(c),
  }))
  return (
    <PeachScrollView contentContainerStyle={[tw`justify-center flex-grow py-sm`, tw.md`py-md`]}>
      <RadioButtons items={currencies} selectedValue={currency} onButtonPress={setCurrency} />
    </PeachScrollView>
  )
}
