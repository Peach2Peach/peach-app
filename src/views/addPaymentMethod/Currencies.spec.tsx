import { Currencies } from './Currencies'
import { createRenderer } from 'react-test-renderer/shallow'

describe('Currencies', () => {
  const renderer = createRenderer()
  const props = {
    currency: 'EUR' as const,
    setCurrency: jest.fn(),
  }

  it('should render correctly', () => {
    renderer.render(<Currencies {...props} />)
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
  })
})
