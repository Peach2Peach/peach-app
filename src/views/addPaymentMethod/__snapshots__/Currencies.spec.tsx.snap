// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currencies should render correctly for type "africa" 1`] = `
<ForwardRef
  contentContainerStyle={
    [
      {
        "flexGrow": 1,
        "justifyContent": "center",
        "paddingBottom": 8,
        "paddingTop": 8,
      },
      {},
    ]
  }
>
  <RadioButtons
    items={
      [
        {
          "display": "US dollar (USD)",
          "value": "USD",
        },
        {
          "display": "West African CFA franc (XOF)",
          "value": "XOF",
        },
        {
          "display": "Naira (NGN)",
          "value": "NGN",
        },
        {
          "display": "Congolese franc (CDF)",
          "value": "CDF",
        },
      ]
    }
    onButtonPress={[MockFunction]}
    selectedValue="EUR"
  />
</ForwardRef>
`;

exports[`Currencies should render correctly for type "europe" 1`] = `
<ForwardRef
  contentContainerStyle={
    [
      {
        "flexGrow": 1,
        "justifyContent": "center",
        "paddingBottom": 8,
        "paddingTop": 8,
      },
      {},
    ]
  }
>
  <RadioButtons
    items={
      [
        {
          "display": "Euro (EUR)",
          "value": "EUR",
        },
        {
          "display": "Swiss franc (CHF)",
          "value": "CHF",
        },
        {
          "display": "British pound (GBP)",
          "value": "GBP",
        },
        {
          "display": "Swedish krona (SEK)",
          "value": "SEK",
        },
        {
          "display": "Danish kroner (DKK)",
          "value": "DKK",
        },
        {
          "display": "Bulgarian lev (BGN)",
          "value": "BGN",
        },
        {
          "display": "Czech koruna (CZK)",
          "value": "CZK",
        },
        {
          "display": "Hungarian forint (HUF)",
          "value": "HUF",
        },
        {
          "display": "Polish złoty (PLN)",
          "value": "PLN",
        },
        {
          "display": "Romanian leu (RON)",
          "value": "RON",
        },
        {
          "display": "Icelandic króna (ISK)",
          "value": "ISK",
        },
        {
          "display": "Norwegian krone (NOK)",
          "value": "NOK",
        },
        {
          "display": "Turkish lira (TRY)",
          "value": "TRY",
        },
      ]
    }
    onButtonPress={[MockFunction]}
    selectedValue="EUR"
  />
</ForwardRef>
`;

exports[`Currencies should render correctly for type "latinAmerica" 1`] = `
<ForwardRef
  contentContainerStyle={
    [
      {
        "flexGrow": 1,
        "justifyContent": "center",
        "paddingBottom": 8,
        "paddingTop": 8,
      },
      {},
    ]
  }
>
  <RadioButtons
    items={
      [
        {
          "display": "Argentine peso (ARS)",
          "value": "ARS",
        },
        {
          "display": "Colombian Peso (COP)",
          "value": "COP",
        },
        {
          "display": "Peruvian sol (PEN)",
          "value": "PEN",
        },
        {
          "display": "Mexican peso (MXN)",
          "value": "MXN",
        },
        {
          "display": "Chilean peso (CLP)",
          "value": "CLP",
        },
        {
          "display": "Peruvian sol (PEN)",
          "value": "PEN",
        },
        {
          "display": "Colombian Peso (COP)",
          "value": "COP",
        },
        {
          "display": "Costa Rican Colón (CRC)",
          "value": "CRC",
        },
        {
          "display": "Brazilian real (BRL)",
          "value": "BRL",
        },
      ]
    }
    onButtonPress={[MockFunction]}
    selectedValue="EUR"
  />
</ForwardRef>
`;

exports[`Currencies should render correctly for type "other" 1`] = `
<ForwardRef
  contentContainerStyle={
    [
      {
        "flexGrow": 1,
        "justifyContent": "center",
        "paddingBottom": 8,
        "paddingTop": 8,
      },
      {},
    ]
  }
>
  <RadioButtons
    items={
      [
        {
          "display": "LNURL BTC swap",
          "value": "SAT",
        },
        {
          "display": "L-USDT",
          "value": "USDT",
        },
      ]
    }
    onButtonPress={[MockFunction]}
    selectedValue="EUR"
  />
</ForwardRef>
`;
